### PR TITLE
feat(operator): catalog metadata types for prior index data

### PR DESCRIPTION
**Pull request by estroz**
This PR adds the ability to store and pull image set metadata from a backend, and fixes some `create` metadata handling issues. Currently only a local directory storage backend is supported.

Signed-off-by: Eric Stroczynski <ericstroczynski@gmail.com>

### CODE REVIEW

**Comment by estroz** [comment]
Blocking on #28 so I can rebase.

File: pkg/bundle/create/create.go [hunk 1]
```diff
@@ -14,40 +15,53 @@ import (
...
-	if len(metadata.PastMirrors) != 0 {
-		return config.NewFullMetadataError(metadataPath)
+	// Read in current metadata
+	meta := v1alpha1.Metadata{}
+	switch err := backend.ReadMetadata(ctx, &meta); {
+	case err != nil && !errors.Is(err, storage.ErrMetadataNotExist):
+		return err
+	case err == nil && len(meta.PastMirrors) != 0:
+		// No past run(s) are allowed when generating a full imageset.
+		// QUESTION: should the command just log a warning and ignore this metadata instead?
```
Comment (jpower432) [inline pkg/bundle/create/create.go:55]: @estroz Responding to your comment in the code, would it better to prompt the user or would we send a warning when creating a full and destroy the current metadata. @afflom and I talked about this and the user should be able to create a full even if there is existing metadata, but the only area of concern I have about this is the user accidentally running `create full`. Maybe a prompt or a metadata backup? WDYT?

File: pkg/bundle/create/create.go [hunk 2]
```diff
@@ -117,30 +130,36 @@ func CreateFull(configPath, rootDir, outputDir string, dryRun, insecure bool) er
...
 	}
 
-	lastRun := metadata.PastMirrors[len(metadata.PastMirrors)-1]
-
-	// Set timestamp
-	lastRun.Timestamp = int(time.Now().Unix())
+	lastRun := meta.PastMirrors[len(meta.PastMirrors)-1]
+	run := v1alpha1.PastMirror{
+		Sequence:  lastRun.Sequence + 1,
+		Uid:       uuid.New(),
```
Comment (jpower432) [inline pkg/bundle/create/create.go:160]:
```suggestion
		Uid:       lastRun.Uid,
```

**Review by jpower432 — CHANGES_REQUESTED** [review]
LGTM, but needs a change for the diff Uid to maintain the same Uid between runs. We may want to also discuss how we are going to handle new `create full` runs on currently managed data.

File: pkg/bundle/create/create.go (hunk 1 above)
Comment (estroz) [reply pkg/bundle/create/create.go:55]: Prompt sounds fine. I'll create a follow-up.

File: pkg/bundle/create/create.go (hunk 2 above)
Comment (afflom) [reply pkg/bundle/create/create.go:160]: That seems logical

**Review by jpower432 — APPROVED** [review]
/lgtm

**Pull request merged**